presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -392,30 +392,30 @@ static boolean isIndexStream(Stream stream)
         return stream.getStreamKind() == ROW_INDEX || stream.getStreamKind() == DICTIONARY_COUNT || stream.getStreamKind() == BLOOM_FILTER || stream.getStreamKind() == BLOOM_FILTER_UTF8;
     }
 
-    private Map<StreamId, List<HiveBloomFilter>> readBloomFilterIndexes(Map<StreamId, Stream> streams, Map<StreamId, OrcInputStream> streamsData)
+    private Map<Integer, List<HiveBloomFilter>> readBloomFilterIndexes(Map<StreamId, Stream> streams, Map<StreamId, OrcInputStream> streamsData)
             throws IOException
     {
-        ImmutableMap.Builder<StreamId, List<HiveBloomFilter>> bloomFilters = ImmutableMap.builder();
+        ImmutableMap.Builder<Integer, List<HiveBloomFilter>> bloomFilters = ImmutableMap.builder();
         for (Entry<StreamId, Stream> entry : streams.entrySet()) {
             Stream stream = entry.getValue();
             if (stream.getStreamKind() == BLOOM_FILTER) {
                 OrcInputStream inputStream = streamsData.get(entry.getKey());
-                bloomFilters.put(entry.getKey(), metadataReader.readBloomFilterIndexes(inputStream));
+                bloomFilters.put(entry.getKey().getColumn(), metadataReader.readBloomFilterIndexes(inputStream));
```

Comment:
Is it possible to have multiple streams for a given column?

Reply:
If Bloom Filter is enabled for a column Then only one Bloom Filter Stream exists per column.There can not exists more than one streams per column.
I will try to build a test case for this.